src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -1828,5 +1828,111 @@ namespace UiaOperationAbstractionTests
         {
             UiaTupleInCollectionsTest(true /* useRemoteOperations */);
         }
+
+        void GetMetadataValueTest(bool useRemoteOperations)
+        {
+            // Initialize the test application.
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+
+            // Set focus to the display element.
+            auto focusedElement = WaitForElementFocus(L"Display is 0");
+
+            // Initialize the UIA Remote Operation abstraction.
+            const auto cleanup = InitializeUiaOperationAbstraction(useRemoteOperations);
+
+            // Get metadata value operation.
+            {
+                auto operationScope = UiaOperationScope::StartNew();
+
+                // Give this operation a remote context.
+                UiaElement displayElement{ focusedElement };
+                operationScope.BindInput(displayElement);
+
+                // Get metadata value from the element
+                UiaVariant variant = displayElement.GetMetadataValue(UIA_NamePropertyId, UIA_SayAsInterpretAsMetadataId);
+                UiaBool isVariantNull = (variant.IsNull());
+                UiaBool isVariantTypeInt{ false };
+                operationScope.If(!variant.IsNull(), [&]()
+                {
+                    isVariantTypeInt = variant.IsType<UiaInt>();
+                });
+
+                // Return the results of the comparisons.
+                operationScope.BindResult(isVariantNull, isVariantTypeInt);
+                operationScope.Resolve();
+
+                // Because the focused element does not hold any metadata value, the returned variant will not contain actual value.
+                if (useRemoteOperations)
+                {
+                    // When use remote operations, the returned variant will be null.
+                    Assert::AreEqual(true, static_cast<bool>(isVariantNull));
```

Comment:
Why are we getting a different result when we use remote ops?

Reply:
The point of this test is to call `UiaElement::GetMetadataValue` and make sure that this function does not throw.
This test cannot really get a metadata value from the element, because no such metadata exists on that element. If we can get real metadata, this problem will go.

In this case, when no metadata exists, the remote version will return a null `UiaVariant`. However, the local version will return a non-null `UiaVariant`, with the local `VARIANT` value has a `VT_EMPTY` type. We could also tell this difference in the implementation of the `UiaVarint::IsNull()` function, the local version always returns false.

Because of all of that, I have distinguished remote/local behavior in this test.

Reply:
Ah, OK. That makes sense.